### ISSUE TITLE
Check windows compilation

### ISSUE THREAD

**Issue by Nicogene** [robotology/blocktest#20]
See https://github.com/robotology/blocktest/runs/495650621

**Comment by traversaro** [comment]
It seems a case-related issue, I guess.

**Comment by Nicogene** [comment]
It seems it also to me, i will try to compile it on my pc following the same steps and see what happens

**Comment by Nicogene** [comment]
Actually it is a real error, it complains about : https://github.com/robotology/blocktest/blob/master/CMakeLists.txt#L118

Any help is super appreciated

**Comment by traversaro** [comment]
> 
> 
> Actually it is a real error, it complains about : https://github.com/robotology/blocktest/blob/master/CMakeLists.txt#L118
> 
> Any help is super appreciated

What is the purpose of that piece  of code? I guess it is to support use directly from the build directory, as typically done by the `yarp_install` macro, but this version is a bit strange, mixing stuff executed at build time and at generation time. The problem, as you can check if you read the `add_custom_command` documentation ( https://cmake.org/cmake/help/latest/command/add_custom_command.html#build-events ) is that in this directory there is no target  `blockTest`, so it is not possible to add a custom command to a target that does not exists. I guess that that target used to exist, but then it changed name, and that part of code was subject to [code rot](https://en.wikipedia.org/wiki/Software_rot). Probably switching the target to `blocktestcore` could fix this, but I am not sure if we can do that in a parent directory.


If using those files directly from the build cannot be used, we can consider removing it, or actually use some code snippets that work: 
* https://github.com/robotology/yarp/blob/a6fd00f6d5659cfee9766354b91b3096754459f8/cmake/YarpInstallationHelpers.cmake#L363
*  https://github.com/robotology/funny-things/blob/874a20063304107b913ecd791216941a55b10a96/app/CMakeLists.txt#L16

**Comment by Nicogene** [comment]
I almost fixed the windows build in this branch: https://github.com/robotology/blocktest/commits/fix/WinBuild

I get this error 
```
Error	C1128	number of sections exceeded object file format limit: compile with /bigobj	C:\Users\ngenesio\robotology\blocktest\build\blocktest	C:\Users\ngenesio\robotology\blocktest\src\blocktestcore\tables.cpp	1	

```

Never saw in my life, I found on internet that it is fixable confiuring the VS project but I don't understand what is the prblem with `table.cpp`

cc Windows users: @traversaro @pattacini @aerydna

**Comment by Nicogene** [comment]
I fixed with `add_compile_options(-bigobj)`, but I still don't get the cause of the problem

**Comment by traversaro** [comment]
> 
> 
> I fixed with `add_compile_options(-bigobj)`, but I still don't get the cause of the problem

See https://github.com/robotology/human-dynamics-estimation/pull/172 and https://github.com/robotology/human-dynamics-estimation/issues/171 . 

> Never saw in my life, I found on internet that it is fixable confiuring the VS project but I don't understand what is the prblem with table.cpp

There is no problem by itself, it probably is just using a lot of templates and external headers, and that is creating a number of symbols that is too much for the `COFF` file format that is the default one for compilation units in Windows, see https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1128?view=vs-2019 . The `/bigobj` option just changes the default format used for compilation units on Windows, and should be harmless.

**Comment by Nicogene** [comment]
Thanks @traversaro for the explaination, the compilation is fine, the CI is failing but it is not related, it is just some issue in the configuration (zip + `vcpkg install` conflicts)

**Comment by Nicogene** [comment]
The issues I had in the CI should be fixed once https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/pull/14 is merged and a new version of the `robotology-dependencies-vcpkg` is released, since now they include also all the boost stuff.

**Comment by traversaro** [comment]
> 
> 
> The issues I had in the CI should be fixed once [robotology/robotology-superbuild-dependencies-vcpkg#14](https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/pull/14) is merged and a new version of the `robotology-dependencies-vcpkg` is released, since now they include also all the boost stuff.

The new archive should be automatically added to the release https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/releases/tag/v0.2.0 as soon as the relative CI jobs finishes (you can follow it at https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/actions/runs/91045458, it should take  ~3/4 hours).

**Comment by Nicogene** [comment]
Fixed by #29
